src/utils/components/DiskModal/hooks/useEditDiskState.ts

```diff
@@ -0,0 +1,123 @@
+import produce from 'immer';
+
+import { V1DataVolumeTemplateSpec, V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import {
+  getBootDisk,
+  getDataVolumeTemplates,
+  getDisks,
+  getVolumes,
+} from '@kubevirt-utils/resources/vm';
+import { diskTypes } from '@kubevirt-utils/resources/vm/utils/disk/constants';
+import { getDiskDrive, getDiskInterface } from '@kubevirt-utils/resources/vm/utils/disk/selectors';
+import { isEmpty, removeDockerPrefix } from '@kubevirt-utils/utils/utils';
+
+import { DYNAMIC } from '../components/utils/constants';
+import { initialStateDiskForm } from '../utils/constants';
+import { DiskFormState, SourceTypes, VolumeTypes } from '../utils/types';
+
+type UseEditDiskState = (vm: V1VirtualMachine, diskName: string) => DiskFormState;
+
+const getSourceFromDataVolume = (dataVolume: V1DataVolumeTemplateSpec): SourceTypes => {
+  if (dataVolume.spec?.source?.http) return SourceTypes.HTTP;
+  if (dataVolume.spec?.source?.pvc) return SourceTypes.CLONE_PVC;
+  if (dataVolume.spec?.source?.registry) return SourceTypes.REGISTRY;
+  if (dataVolume.spec?.source?.blank) return SourceTypes.BLANK;
+  return SourceTypes.OTHER;
+};
+
+export const useEditDiskState: UseEditDiskState = (vm, diskName) => {
+  const state = produce(initialStateDiskForm, (draftState) => {
```

Comment:
feels like a lot of functions can be split into small parts an exported to util file